### PR TITLE
refactor: iwyu

### PR DESCRIPTION
**Pull request by justinmk**
Result of `make iwyu` (after some "fixups").

### CODE REVIEW

File: src/nvim/tui/termkey/termkey.c [hunk 1]
```diff
@@ -13,9 +13,7 @@
 #include "nvim/tui/termkey/termkey_defs.h"
 
 #ifndef _WIN32
-# include <poll.h>
-# include <strings.h>
-# include <unistd.h>
+# include <termios.h>
```
Comment (justinmk) [inline src/nvim/tui/termkey/termkey.c:16]: 👀

File: src/nvim/context.h [hunk 2]
```diff
@@ -1,7 +1,5 @@
 #pragma once
 
-#include <stddef.h>
```
Comment (zeertzjq) [inline src/nvim/context.h:3]: This is needed by the generated header

File: src/nvim/api/private/helpers.h [hunk 3]
```diff
@@ -1,7 +1,6 @@
 #pragma once
 
 #include <stdbool.h>
-#include <stddef.h>
```
Comment (zeertzjq) [inline src/nvim/api/private/helpers.h:4]: ditto

File: src/nvim/cmdhist.h [hunk 4]
```diff
@@ -1,7 +1,8 @@
 #pragma once
 
+#include <stddef.h>
+
 #include "nvim/cmdexpand_defs.h"  // IWYU pragma: keep
-#include "nvim/eval/typval_defs.h"
```
Comment (zeertzjq) [inline src/nvim/cmdhist.h:4]: ditto

File: src/nvim/option.h [hunk 5]
```diff
@@ -1,13 +1,10 @@
 #pragma once
 
-#include <stdbool.h>
-#include <stdint.h>
 #include <stdio.h>  // IWYU pragma: keep
 
 #include "nvim/api/private/defs.h"  // IWYU pragma: keep
 #include "nvim/api/private/helpers.h"
 #include "nvim/cmdexpand_defs.h"  // IWYU pragma: keep
-#include "nvim/eval/typval_defs.h"
```
Comment (zeertzjq) [inline src/nvim/option.h:10]: ditto

File: src/nvim/context.h (hunk 2 above)
Comment (justinmk) [reply src/nvim/context.h:3]: Thanks :) If IWYU doesn't see that, and the build passes... hmm...

**Pull request merged**